internal/docker/run.go

```diff
@@ -0,0 +1,66 @@
+package docker
+
+import (
+	"fmt"
+	"io"
+
+	mobycontainer "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+)
+
+// RunOptions defines the container to create and start.
+type RunOptions struct {
+	// Pull will ensure the image exists locally before running.
+	//
+	// If an image will only be used once, this is a convenience to avoid calling ImagePull first.
+	// If an image will be used multiple times (across containers), prefer explicitly calling ImagePull
+	// to avoid the overhead of calling the registry API to check if the image is up-to-date every time.
+	Pull bool
+	// ContainerName is a unique name for the container. If not specified, Docker will generate a random name.
+	ContainerName string
+	// ContainerConfig is the main configuration for the container.
+	//
+	// The Image field MUST be populated at a minimum.
+	ContainerConfig mobycontainer.Config
```

Comment:
it might make sense to have narrower settings than the whole Config?

i feel like there are settings here that are incompatible with how you're running the container (e.g., AutoCleanup and RestartPolicy)

Reply:
Fair - the Docker types are giant, so I was a little afraid of having an obnoxious amount of mapping code, but I think realistically Tilt will only ever want/need to use a handful of options - refactored in f758868